src/plugins/diagram-viewer/diagram-tool.tsx

```diff
@@ -14,12 +15,26 @@ export const DiagramToolComponent: React.FC<IToolTileProps> = (
 ) => {
   const content = model.content as DiagramContentModelType;
 
+  const [diagramDialogOpen, setDiagramDialogOpen] = useState(false);
+  const [showDiagramDialog, hideDiagramDialog] = useDiagramDialog({
+    onAccept: () => console.log("Acceptable."),
+    onClose: () => setDiagramDialogOpen(false)
+  });
+  useEffect(() => {
+    if (diagramDialogOpen) {
+      showDiagramDialog();
+    } else {
+      hideDiagramDialog();
+    }
+  }, [diagramDialogOpen, hideDiagramDialog, showDiagramDialog]);
+
   const toolbarProps = useToolbarToolApi({ id: model.id, enabled: !readOnly, onRegisterToolApi, onUnregisterToolApi });
   
   return (
     <div className="diagram-tool">
       <DiagramToolbar
         documentContent={documentContent}
+        handleDialogClick={() => setDiagramDialogOpen(true)}
```

Comment:
Is it necessary to use a state variable here? Could you just call `showDialogDialog()`?